### PR TITLE
Account activation/deactivation

### PR DESCRIPTION
**Pull request by jcarrete5**
Not really sure what else is needed for this. The documentation for [User.is_active](https://docs.djangoproject.com/en/3.1/ref/contrib/auth/#django.contrib.auth.models.User.is_active) seems like a different use case than what we want. I'm not sure what we want.

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -302,7 +302,8 @@ def post(self, request, *args, **kwargs):
         return self.create(request, *args, **kwargs)
 
 
-class Accounts(generics.RetrieveAPIView):
-    def get(self, request):
-        serializer = AccountSerializer(request.user)
-        return response.Response(serializer.data, status=status.HTTP_200_OK)
+class Accounts(generics.RetrieveUpdateAPIView):
```
Comment (rao53) [inline api/views.py]: Much cleaner!

**Review by rao53 — CHANGES_REQUESTED** [review]
Looks good. Some frontend things to note are:
- User isn't logged out after deactivation.
- How do I reactivate my account if I can't log in to reactivate it again?

**Review by AyalaCnaan — COMMENTED** [review]
Would really like to see a screenshot of what this looks like.

**Review by mneedleman — CHANGES_REQUESTED** [review]
Like Ridwan pointed out, it looks like this is missing the method for reactivation

**Review by elp49 — DISMISSED** [review]

**Review by mneedleman — DISMISSED** [review]

**Review by elp49 — DISMISSED** [review]

**Comment by AyalaCnaan** [comment]
> Would really like to see a screenshot of what this looks like.

Would still like to see what this will look like step by step.

**Comment by jcarrete5** [comment]
> > Would really like to see a screenshot of what this looks like.
> 
> Would still like to see what this will look like step by step.

Sorry for late response. The account UI is unchanged, only confirmation dialogs were added before deactivation after attempting a login with a paused account.

Pausing an account
image
Reactivating an account after a failed login
image

**Review by AyalaCnaan — DISMISSED** [review]
Other than the utterly horrible pink of the buttons that clashes with our entire color scheme in the confirmation modals, this is fine with me.

**Comment by jcarrete5** [comment]
> Other than the utterly horrible pink of the buttons that clashes with our entire color scheme in the confirmation modals, this is fine with me.

Hopefully we get around to getting our MUI theme sorted and consistent before we record our demo.

**Review by rao53 — DISMISSED** [review]

File: api/models.py [hunk 2]
```diff
@@ -136,10 +140,15 @@ class ReportType(models.TextChoices):
         on_delete=models.SET_NULL,
         limit_choices_to={"is_staff": True},
         null=True,
+        blank=True,
         help_text="The admin handling the report",
         related_name="assigned_reports",
     )
     video = models.ForeignKey(
```
Comment (rao53) [inline api/models.py:147]: Was this the change from master or you made this change?

**Review by rao53 — COMMENTED** [review]
^^

File: api/models.py (hunk 2 above)
Comment (jcarrete5) [reply api/models.py:147]: No. Github is a bit weird when showing the changes from a merge. Click on the latest commit hash to see the changes I made to fix the merge conflicts.

Comment (jcarrete5) [reply api/models.py:147]: The change was made in this PR but not to fix the merge conflict.

**Review by mneedleman — APPROVED** [review]

**Review by rao53 — APPROVED** [review]

**Pull request merged**